float32 x
float32 y
float32 concentration_1
float32 concentration_2
float32 x_increasement
float32 y_increasement
float32 w